enterprise/views.py

```diff
@@ -1698,22 +1700,47 @@ def post(self, request, enterprise_uuid, course_id):
             # For the audit course modes (audit, honor), where DSC is not
             # required, enroll the learner directly through enrollment API
             # client and redirect the learner to LMS courseware page.
-            if not enterprise_course_enrollment:
-                # Create the Enterprise backend database records for this course enrollment.
-                EnterpriseCourseEnrollment.objects.create(
-                    enterprise_customer_user=enterprise_customer_user,
-                    course_id=course_id,
-                    source=enrollment_source
-                )
-                track_enrollment('course-landing-page-enrollment', request.user.id, course_id, request.get_full_path())
-
+            succeeded = True
             client = EnrollmentApiClient()
-            client.enroll_user_in_course(
-                request.user.username,
-                course_id,
-                selected_course_mode_name,
-                cohort=cohort_name
-            )
+            try:
+                client.enroll_user_in_course(
+                    request.user.username,
+                    course_id,
+                    selected_course_mode_name,
+                    cohort=cohort_name
+                )
+            except HttpClientError as exc:
+                succeeded = False
+                default_message = 'No error message provided'
```

Comment:
To be honest, I'm not sure! I decided to just continue using the convention we have in place for handling these HttpClientErrors elsewhere, which gracefully handles when an HttpClientError comes back without a message.

Figured it'd be safer to continue using what's in place than reinvent the wheel here. That said, the message "No error message provided" hasn't appeared in our Splunk logs at least in the past 7 days so I agree these HttpClientErrors are likely always coming back with a message attached.

I feel it makes sense to keep the exception handling defensive juuust in case there's some error while trying to parse the exception content.

Reply:
Works for me, better safe than sorry. Especially with something as important and enrollments.